workflows/nimble_workflows.py

```diff
@@ -0,0 +1,311 @@
+        snap = client.snapshots.get(id=snap_id)
+        if snap is None:
+            screen('ERROR: Snapshot does not exist: {}'.format(snap_id), noisy)
+            return
+        # Delete snapshot
+        client.snapshots.delete(id=snap_id)
+        screen('\tDeleted snapshot: {}, Id: {}'.format(snap.attrs['name'], snap.id), noisy)
+    except NimOSAPIError:
+        traceback.print_stack()
+        traceback.print_exc()
+
+
+def clone_volume(client, vol_name, noisy):
+    try:
+        screen('\nWorkflow: Clone Volume - {}'.format(vol_name), noisy)
+        snap_id = create_snapshot(client, vol_name, noisy=False)
+        clone_name = vol_name + 'clone'
+        clone = client.volumes.create(name=clone_name, base_snap_id=snap_id, clone=True)
+        screen('\tCloned volume: {}, Id: {}'.format(clone.attrs['name'], clone.id), noisy)
+        clone_acr_id = publish_volume(client, clone.attrs['id'], noisy=False)
+        screen('\tPublished cloned volume: {}'.format(clone_name), noisy)
+        detach_volume(client, clone_acr_id, noisy=False)
+        screen('\tDeleted cloned volume: {}'.format(clone_name), noisy)
+    except NimOSAPIError:
+        traceback.print_stack()
+        traceback.print_exc()
+
+
+def protect_volume(client, vol_id, noisy):
+    try:
+        screen('\nWorkflow: Protect Volume - {}'.format(vol_id), noisy)
+        # Get the volume
+        vol = client.volumes.get(vol_id)
+        # Create a volume volume_collection
+        volcoll_name = vol.attrs['name'] + 'coll'
+        volcoll1 = client.volume_collections.create(name=volcoll_name, description="created by workflow")
+        screen('\tCreated volume collection: {}, Id: {}'.format(volcoll1.attrs['name'], volcoll1.id), noisy)
+        # Associate a volume with the volume collection
+        asoc = client.volumes.associate(id=vol.attrs['id'], volcoll=volcoll1)
+        screen('\tAssociated volume: {}, with volume vollection: {}'.format(
+               asoc['access_control_records'][0]['vol_name'], asoc['volcoll_name']), noisy)
+        # Create a protection schedule
+        ps_name = "ps1"
+        ps_days = "monday,tuesday,wednesday,thursday,friday"
+        ps_info = "just a schedule"
+        ps = client.protection_schedules.create(name=ps_name, period=60, period_unit="minutes",
+                                                days=ps_days, description=ps_info,
+                                                volcoll_or_prottmpl_id=volcoll1.id,
+                                                volcoll_or_prottmpl_type='volume_collection',
+                                                num_retain=2)
+        screen('\tCreated protection schedule: {}, Id: {}'.format(ps.attrs['name'], ps.id), noisy)
+        client.protection_schedules.delete(id=ps.attrs['id'])
+        screen('\tDeleted protection schedule: {}'.format(ps.attrs['name']), noisy)
+        # Disssociate a volume with the volume collection and delete to volume collection
```

Comment:
delete the volume collection

Reply:
This happens on line 233, below.

Reply:
I was pointing out a grammatical issue:

> and delete to volume collection

should be "and delete **the** volume collection"

Reply:
Gotcha :)